### PR TITLE
Features/recipe repository class

### PR DESCRIPTION
**Pull request by laytonmaes**
# Description
refactored the recipe repository class, setting it up to have a method to iterate through a data set and build class instances of the recipes, as well as filtering out all individual tags. it also has a function for searching by tag and an additional method for filtering by user input.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x ] Refactoring
## How Has This Been Tested?
- [ ] open localHost
- [x] dev tools
## Issues
#1 
## Checklist:
- [x] My code follows the Turing’s guidelines of this project
- [x] I have performed a self-review of my own code
- [x] No console error messages

### CODE REVIEW

File: src/classes/RecipeRepository.js [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
   }
   
+  addRecipes() {
+    this.recipeObjects = recipeData.map((data) => {
+      return new Recipe(data)
+    })
+  }
+
+  getTags() {
+    this.tags = recipeData.reduce((acc, recipe) => {
```
Comment (angelewilliams) [inline src/classes/RecipeRepository.js:21]: Glad we used reduce here!

File: src/classes/User.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const {ingredientsData} = require('../data/ingredients');
```
Comment (angelewilliams) [inline src/classes/User.js:1]: Nice to have this set up for next time!

File: test/RecipeRepository-test.js [hunk 3]
```diff
@@ -43,7 +56,10 @@ describe('Recipe Repository', () => {
     expect(output2.length).to.equal(6);
     expect(output2[5].id).to.equal(583738);
   });
-
+  it("should be able to record all individual tags from the data set", () => {
```
Comment (angelewilliams) [inline test/RecipeRepository-test.js:59]: Super sweet that we have the list of tags now!

**Review by angelewilliams — APPROVED** [review]

File: src/classes/RecipeRepository.js [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+import Ingredient from './Ingredient';
+import Recipe from './Recipe';
+
 class RecipeRepository {
-  constructor(recipeData) {
-    this.recipeObjects = recipeData;
+  constructor() {
+    this.recipeObjects = [];
+    this.tags = [];
     // One class to get you started!
```
Comment (michaelputnam67) [inline src/classes/RecipeRepository.js:10]: Remove this commented out code #6

File: test/RecipeRepository-test.js [hunk 5]
```diff
@@ -1,23 +1,33 @@
 import { expect } from 'chai';
 import RecipeRepository from '../src/classes/RecipeRepository';
+import Recipe from "../src/classes/Recipe";
 const {recipeData} = require('../src/data/recipes');
 
 
 
 // const repository = new RecipeRepository(recipeData);
```
Comment (michaelputnam67) [inline test/RecipeRepository-test.js:8]: #6

**Review by michaelputnam67 — COMMENTED** [review]
Looks great! satisfied with our progress!

**Pull request merged**